jvm/src/test/scala/mouse/StringJvmTests.scala

```diff
@@ -38,4 +41,30 @@ class StringJvmTests extends MouseSuite {
     "invalid uri".parseURI should ===(new URISyntaxException("invalid uri", "Illegal character in path at index 7").asLeft)
   }
 
+  test("parseUUID") {
+    val validUUIDStr = "00000000-0000-0000-0000-000000000000"
+    val invalidUUIDStr = "invalid"
+
+    validUUIDStr.parseUUID should ===(UUID.fromString(validUUIDStr).asRight[IllegalArgumentException])
+    validUUIDStr.parseUUID.toValidated should ===(validUUIDStr.parseUUIDValidated)
+    validUUIDStr.parseUUID.toOption should ===(validUUIDStr.parseUUIDOption)
+
+    invalidUUIDStr.parseUUID should ===(new IllegalArgumentException("Invalid UUID string: invalid").asLeft[UUID])
+    invalidUUIDStr.parseUUID.toValidated should ===(invalidUUIDStr.parseUUIDValidated)
+    invalidUUIDStr.parseUUID.toOption.isEmpty shouldBe true
+
+    val uuidStringGen: Gen[String] = Arbitrary.arbUuid.arbitrary.map(_.toString)
```

Comment:
Im not convinced the use of Scalacheck is justified. Are we just testing the java UUID library does what it says?

Please omit this section, for lower test complexity.

Reply:
Looks like a fairly standard equivalence check: does parsing a UUID here do the same as the Java UUID library?

Reply:
Yeah, it was over the top, I removed it.